### PR TITLE
Versioned namespaces

### PR DESCRIPTION
**Pull request by jakedt**

### CODE REVIEW

**Comment by jakedt** [comment]
Note: this make may performance somewhat less predictable, but fixes a real issue with the rollout of new schemas. Now the schema version that was in place as of the chose query revision will be the one that is used to calculate the answer uniformly across all nodes.

**Review by ecordell — DISMISSED** [review]
LGTM

File: internal/datastore/memdb/memdb.go [hunk 1]
```diff
@@ -406,3 +441,10 @@ func createNewTransaction(txn *memdb.Txn) (uint64, error) {
 
 	return newTransactionID, nil
 }
+
+func filterToLiveObjects(revision datastore.Revision) memdb.FilterFunc {
+	return func(hasLifetimeRaw interface{}) bool {
+		obj := hasLifetimeRaw.(hasLifetime)
+		return uint64(revision.IntPart()) < obj.getCreatedTxn() || uint64(revision.IntPart()) >= obj.getDeletedTxn()
```
Comment (josephschorr) [inline internal/datastore/memdb/memdb.go:448]: Mind adding a comment here that the filter is inverted?

**Review by josephschorr — APPROVED** [review]
LGTM

**Pull request merged**